crates/turborepo-lib/src/run/summary/spaces.rs

```diff
@@ -257,8 +272,14 @@ mod tests {
             "rauchg".to_string(),
         ))?;
 
+        let mut join_set = tokio::task::JoinSet::new();
         for task_summary in tasks {
-            spaces_client_handle.finish_task(task_summary).await?;
+            let task_client = spaces_client_handle.task_client();
+            join_set.spawn(async move { task_client.finish_task(task_summary).await });
```

Comment:
I'm confused, is there a reason we're not passing `task_client.finish_task(task_summary)` directly?